schemas/simple-bus.yaml

```diff
@@ -55,4 +57,9 @@ required:
   - compatible
   - "#address-cells"
   - "#size-cells"
-  - ranges
+
+anyOf:
+  - required:
+      - dma-ranges
+  - required:
+      - ranges
```

Comment:
ranges is always required and dma-ranges is optional. This change is not needed.

Reply:
How to describe the stm32mp1 coprocessor mapping and translations based on this requirement?
On stm32mp1 we have 2 RAM memories accessible by the Cortex-A7 running Linux and used by the Cortex-M4 coprocessor to run it code. each Cortex have a specific mapping of the memories. Some memory translations are needed by:
- rproc driver to load the Cortex-M4 firmware in RAMs
- rpmsg & virtio frameworks for shared memory

Here is the memory mapping:
- RETRAM (64 kB);
  - Cortex-A7 @ : 0x38000000
  - Cortex-M4 @: 0x00000000

- MCUSRAM (up to 384 kB)
  - Cortex-A7 @ : 0x30000000 or 0x10000000 ( 2 addresses for the same memory)
  - Cortex-M4 @: 0x30000000 or 0x10000000 ( 2 addresses for the same memory)
  =>addresses used on both side depend on the MCURAM access optimization

Today our upstreamed solution is based on dma-range only
https://elixir.bootlin.com/linux/v5.4/source/arch/arm/boot/dts/stm32mp157c.dtsi

What about adding an "unused" ranges property in DT to match the requirement?
```
	mlahb {
		compatible = "simple-bus";
		#address-cells = <1>;
		#size-cells = <1>;
		ranges = <0 0x38000000 0x10000>,
			       <1 0x10000000 0x60000>,
			       <2 0x30000000 0x60000>;
		dma-ranges = <0x00000000 0x38000000 0x10000>,
			     <0x10000000 0x10000000 0x60000>,
			     <0x30000000 0x30000000 0x60000>;

		m4_rproc: m4@10000000 {
		        ---
		};
	};
```

Thanks in advance for your feedback.
Regards,
Arnaud

Reply:
If dma-ranges is required, then you need a more specific compatible for the node than 'simple-bus'. Please respin this to only add 'dma-ranges: true'

Reply:
i think this PR can be closed without change, 
Point has been treated in https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/arm/stm32/st,mlahb.yaml